update stict mode language (#8813)
fishtown-analytics/docs.getdbt.com: website/docs/docs/fusion/new-concepts.md

## Configuring `static_analysis`

You can modify the way static analysis is applied for specific models in your project. The static analysis configuration cascades from most strict to least strict. Going downstream in your lineage, a model can keep the same mode or relax it &mdash; it can't be stricter than its parent. 

> Setting a model to `strict` does not automatically make its downstreams strict; they keep the project default unless you set them explicitly. For the full rules and examples, see [How static analysis modes cascade](/reference/resource-configs/static-analysis#how-static-analysis-modes-cascade).

Author: ```suggestion
Setting `static_analysis: strict` on a model does not automatically set `strict` for downstream models; they keep the project default unless you set them explicitly. For rules and examples, refer to [strict mode inheritance](#strict-mode-inheritance) and [How static analysis modes cascade](/reference/resource-configs/static-analysis#how-static-analysis-modes-cascade).
```

Reviewer: Hiya Mirna

Just amended this slightly to tighten the wording and make the language ("see") a bit more accessible.

</File>

#### strict mode inheritance

> Unlike `baseline` or `off`, `strict` mode doesn't propagate to downstream models. This means that if you set a model to `strict`, its downstream models will not inherit `strict` mode unless you set them explicitly. To make all models `strict`, you must set `+static_analysis: strict` on root models first, or use the project-wide config in the next section at the project level.

Reviewer: ```suggestion
Unlike `baseline` or `off`, `strict` mode doesn't propagate to downstream models. If you configure a model as `strict`, its downstream models remain at the project default unless you configure them explicitly. To make all models `strict`, start by setting `+static_analysis: strict` on root models (or use the project-wide configuration in the next section).
```

Author: thanks @nataliefiann, great feedback! 

i prefer being a bit more explicit by saying the downstream models wont inherit strict mode rather than saying the downstream models remain at the project default (which then forces the user to figure out what the default is). so i'll meet you halfway and keep the first sugg but keep the rest.

> Unlike `baseline` or `off`, `strict` mode doesn't propagate to downstream models. This means that if you set a model to `strict`, its downstream models will not inherit `strict` mode unless you set them explicitly. To make all models `strict`, you must set `+static_analysis: strict` on root models first, or use the project-wide config in the next section at the project level.

Author: ```suggestion
Unlike `baseline` or `off`, `strict` mode doesn't propagate to downstream models. If you configure a model as `strict`, its downstream models won't inherit `strict` mode unless you set them explicitly. To make all models `strict`, you must set `+static_analysis: strict` on root models first, or use the project-wide config in the next section at the project level.
```

Author: how about this @nataliefiann ?

> For example, in A → B → C with a default of `baseline`, setting `strict` on A (a root node) makes only A `strict` &mdash; B and C remain `baseline` unless configured. To make the full chain strict, set `+static_analysis: strict` on each relevant model or group, or use a project-wide setting.

Reviewer: ```suggestion
For example, in A → B → C with a default of `baseline`, configuring A (a root node) as `strict` makes only A `strict` &mdash; B and C remain baseline unless configured. To make the full chain `strict`, set `+static_analysis: strict` on each relevant model or group, or use a project-wide setting.
```

Reviewer: Hiya Mirna

Just amended this a tad as to tighten it.

This approach lets you gain the benefits of strict validation where possible while keeping the flexibility of baseline analysis for models that aren't yet compatible.

Refer to [CLI options](/reference/global-configs/command-line-options) and [Configurations and properties](/reference/configs-and-properties) to learn more about configs.

### Example configurations

##### Configure strict for the entire project

> Many teams want to enable strict mode for the whole project and all packages. You can do this by setting `+static_analysis: strict` under each resource type in `dbt_project.yml` for your project name (and for any package names if you want those to be strict too):

Reviewer: ```suggestion
Many teams want to enable `strict` mode for the whole project and all packages. You can do this by setting `+static_analysis: strict` under each resource type in `dbt_project.yml` for your project name (and for any package names if you want those to be strict too):
```

<File name='dbt_project.yml'>

```yaml
models:
  my_project:
    +static_analysis: strict

seeds:
  my_project:
    +static_analysis: strict

snapshots:
  my_project:
    +static_analysis: strict

tests:
  my_project:
    +static_analysis: strict

unit_tests:
  my_project:
    +static_analysis: strict

sources:
  my_project:
    +static_analysis: strict

analyses:
  my_project:
    +static_analysis: strict
```

</File>

> Use your project name in place of `my_project` — that's the same value as the `name:` key at the top of `dbt_project.yml` (for example, `jaffle_shop`). To apply strict to a package as well, add another entry under each resource type using the package name as the key; for example, under `models:` add `your_package_name:` with `+static_analysis: strict` beneath it.

Reviewer: ```suggestion
Use your project name in place of `my_project` — that's the same value as the `name:` key at the top of `dbt_project.yml` (for example, `jaffle_shop`). To apply `strict` to a package as well, add another entry under each resource type using the package name as the key; for example, under `models:` add `your_package_name:` with `+static_analysis: strict` beneath it.
```
